src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverterTest.java

```diff
@@ -0,0 +1,62 @@
+package io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+import com.datastax.oss.driver.api.core.data.CqlDuration;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class CqlDurationConverterTest {
+  @ParameterizedTest
+  @MethodSource("validCqlDurationToISO8601Duration")
+  void testCqlToISO8601Duration(String input, String expectedISO8601) {
+    final CqlDuration cqlDuration =
+        assertDoesNotThrow(
+            () -> CqlDuration.from(input),
+            String.format("Converting from String '%s' to CqlDuration", input));
+    String iso8601String =
+        assertDoesNotThrow(
+            () -> CqlDurationConverter.toISO8601Duration(cqlDuration),
+            String.format(
+                "Converting from '%s' (source '%s') to ISO-8601 String'", cqlDuration, input));
+    assertThat(iso8601String)
+        .as("Converting from '%s' to ISO-8601 String, expecting '%s'", input, expectedISO8601)
+        .isEqualTo(expectedISO8601);
+  }
+
+  private static Stream<Arguments> validCqlDurationToISO8601Duration() {
+    // Arguments: (input-to-cql-duration, expected-output)
+    return Stream.of(
+        // Identity
+        Arguments.of("P1Y2M3DT4H5M6S", "P1Y2M3DT4H5M6S"),
+
+        // Actual conversions; manual examples:
+        Arguments.of("1h30m", "PT1H30M"),
+        Arguments.of("3h7s", "PT3H7S"),
+        Arguments.of("22h1ms", "PT22H0.001S"),
+        Arguments.of("1m37us", "PT1M0.000037S"),
+
+        // Actual conversions; inspired by ChatGPT:
```

Comment:
This is one of the moments that we are glad of having AI. 😊

Reply:
Indeed. I did have to fix couple of small things, but overall it helped.